src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,261 @@
+            = "Can't get driver for car by car id = ";
+    private static final String CANT_DELETE_DRIVERS_FROM_CAR_MESSAGE
+            = "Can't delete drivers from car ";
+    private static final String CANT_GET_ALL_CARS_OF_DRIVER_MESSAGE
+            = "Can't get all cars with driverId = ";
+    private static final String NAME = "name";
+    private static final String LICENSE_NUMBER = "license_number";
+    private static final String ID = "id";
+    private static final String CAR_ID = "car_id";
+    private static final String CAR_NAME = "car_name";
+    private static final String MANUFACTURER_ID = "manufacturer_id";
+    private static final String MANUFACTURER_NAME = "manufacturer_name";
+    private static final String MANUFACTURER_COUNTRY = "manufacturer_country";
+
+    @Override
+    public Car create(Car car) {
+        String createCarRequest
+                = "INSERT INTO cars(name, manufacturer_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createCarStatement =
+                         connection.prepareStatement(createCarRequest,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getName());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(String.format(CANT_CREATE_CAR_MESSAGE, car), e);
+        }
+        insertDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarRequest = "SELECT"
+                + " cars.id AS car_id,"
+                + " cars.name AS car_name,"
+                + " manufacturers.id AS manufacturer_id,"
+                + " manufacturers.country AS manufacturer_country,"
+                + " manufacturers.name AS manufacturer_name,"
+                + " manufacturers.country"
+                + " FROM cars JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + " WHERE cars.id = ? AND cars.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getCarStatement
+                         = connection.prepareStatement(getCarRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(String.format(CANT_GET_CAR_MESSAGE, id), e);
```

Comment:
Why `RuntimeException`? We should use our custom `DataProcessingException`(fix in all places)

Reply:
fixed